### PR TITLE
feat: add erc20 deployment scripts

### PR DESCRIPTION
**Pull request by aon**
# What ❔

- Add ERC20 deployment scripts

## Why ❔

- Continue migrating from hardhat to foundry scripts

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.

### CODE REVIEW

File: l1-contracts-foundry/script/DeployErc20.s.sol [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        uint256 mint
+    ) internal returns (address) {
+        bytes memory args = abi.encode(name, symbol, decimals);
+        bytes memory bytecode = abi.encodePacked(vm.getCode(implementation), args);
+
+        address tokenAddress;
+        vm.broadcast();
+        assembly {
+            tokenAddress := create(0, add(bytecode, 0x20), mload(bytecode))
+        }
```
Comment (Deniallugo) [inline l1-contracts-foundry/script/DeployErc20.s.sol:101]: why not reuse create2?

Comment (aon) [reply l1-contracts-foundry/script/DeployErc20.s.sol:101]: I'm migrating the code from hard hat scripts, where normal Create deployment is used. If there's no particular reason this is used, then I'll migrate to Create2

Comment (Deniallugo) [reply l1-contracts-foundry/script/DeployErc20.s.sol:101]: just some old stuff

File: l1-contracts-foundry/script/RegisterHyperchain.s.sol [hunk 2]
```diff
@@ -272,17 +272,17 @@ contract RegisterHyperchainScript is Script {
...
             config.contracts.baseTokenGasPriceMultiplierNominator,
             config.contracts.baseTokenGasPriceMultiplierDenominator
         );
 
-        if (config.contracts.validiumMode) {
-            zkSyncStateTransition.setValidiumMode(PubdataPricingMode.Validium);
-        }
+        // if (config.contractsMode) {
+        //     zkSyncStateTransition.setValidiumMode(PubdataPricingMode.Validium);
+        // }
```
[the remark this answers is not in the document]
Comment (Deniallugo) [reply l1-contracts-foundry/script/RegisterHyperchain.s.sol:285]: Add todo for supporting validium mode

File: l1-contracts-foundry/script/InitializeL2WethToken.sol [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        vm.broadcast();
+        bridgehub.requestL2TransactionDirect{value: requiredValueToInitializeBridge}(l2TransactionRequestDirect);
+
+        console.log("L2 WETH token initialized");
+    }
+
+    function getL2Calldata() internal returns (bytes memory) {
+        // solhint-disable-next-line func-named-parameters
+        bytes memory upgradeData = abi.encodeWithSignature(
+            "initializeV2(string,string,address,address)",
```
Comment (Deniallugo) [inline l1-contracts-foundry/script/InitializeL2WethToken.sol:107]: Please add comments, why is it hardcoded

**Review by Deniallugo — APPROVED** [review]

File: l1-contracts-foundry/script/InitializeL2WethToken.sol [hunk 4]
```diff
@@ -69,12 +69,13 @@ contract InitializeL2WethTokenScript is Script {
         config.l2WethTokenImplAddr = toml.readAddress("$.l2_weth_token_impl_addr");
         config.deployL2BridgeCounterpartGasLimit = toml.readUint("$.deploy_l2_bridge_counterpart_gas_limit");
         config.requiredL2GasPricePerPubdata = toml.readUint("$.required_l2_gas_price_per_pubdata");
+        config.gasMultiplier = toml.readUint("$.gas_multiplier");
     }
 
     function initializeL2WethToken() internal {
         Bridgehub bridgehub = Bridgehub(config.bridgehubProxyAddr);
 
-        uint256 gasPrice = Utils.bytesToUint256(vm.rpc("eth_gasPrice", "[]"));
```
Comment (Deniallugo) [inline l1-contracts-foundry/script/InitializeL2WethToken.sol]: why do we need it?

File: l1-contracts-foundry/script-config/config-initialize-l2-weth-token.toml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+l2_shared_bridge_addr = "0x0000000000000000000000000000000000000000"
+l2_weth_token_proxy_addr = "0x0000000000000000000000000000000000000000"
+l2_weth_token_impl_addr = "0x0000000000000000000000000000000000000000"
+deploy_l2_bridge_counterpart_gas_limit = "10000000"
+required_l2_gas_price_per_pubdata = "800"
+gas_multiplier = "3"
```
Comment (Deniallugo) [inline l1-contracts-foundry/script-config/config-initialize-l2-weth-token.toml:6]: Seems it's not used anymore

File: l1-contracts-foundry/script/InitializeL2WethToken.sol [hunk 6]
```diff
@@ -69,12 +69,13 @@ contract InitializeL2WethTokenScript is Script {
...
+        uint256 gasPrice = Utils.bytesToUint256(vm.rpc("eth_gasPrice", "[]")) * config.gasMultiplier;
         uint256 requiredValueToInitializeBridge = bridgehub.l2TransactionBaseCost(
             config.eraChainId,
             gasPrice,
```
Comment (aon) [inline l1-contracts-foundry/script/InitializeL2WethToken.sol]: It's used here @Deniallugo

File: l1-contracts-foundry/script-config/config-initialize-l2-weth-token.toml (hunk 5 above)
Comment (aon) [reply l1-contracts-foundry/script-config/config-initialize-l2-weth-token.toml:6]: Which parameter is not being used in particular?

**Pull request merged**